projects/igniteui-angular/grids/pivot-grid/src/pivot-grid.component.ts

```diff
@@ -991,6 +991,11 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
                 this.generateConfig();
             }
             this.setupColumns();
+            // Bind to onResourceChange after the columns have initialized the first time to avoid premature initialization.
+            onResourceChangeHandle(this.destroy$, () => {
+                // Since the columns are kinda static, due to assigning DisplayName on init, they need to be regenerated.
```

Comment:
Is this because of the predefined dimensions like the `IgxPivotDateDimension` that have localization? If so it might be a bit of an overkill to regenerate all columns, especially since there might not be a date dimension to localize for.

Reply:
Well localization shouldn't be changed often. The other solution would require some updates to how dimension columns are created and way to sync them with their corresponding dimensions and values.
Sounds like a good improvement, don't think this is the place to do.